### ISSUE TITLE
CI: Warning `Unable to restore router IP from filesystem` in WireGuard tests

### ISSUE THREAD

**Issue by pchaigno** [cilium/cilium#29715]
Happened in https://github.com/cilium/cilium/actions/runs/7132840574/job/19424351381
```
2023-12-07T19:23:01.902716091Z level=warning msg="Unable to restore router IP from filesystem" error="provided IP is not in the valid range. The range of valid IPs is 10.244.0.0/24" ipAddr=10.0.1.128 subsys=daemon
2023-12-07T19:23:01.902719226Z level=warning msg="Router IP could not be re-allocated. Need to re-allocate. This will cause brief network disruption" subsys=daemon
```

This is caused by the `WireGuard encryption strict mode` tests, which appear to change the cluster pool pod CIDR. It's unclear to me if it could be avoided. Maybe by cleaning up old endpoints before reinstalling?

Warning currently allowlisted at https://github.com/cilium/cilium/blob/5a82225cd713ed4480324a3ebfce449196cbbd3c/test/helpers/cons.go#L248C1-L248C41.

cc @brb @gandro

**Comment by gandro** [comment]
Yeah, I think this is expected. The test is doing some intentionally disruptive things.

Short-term, if we want to avoid the warning, I think the options are to either expose `option.Config.EnableHostIPRestore` via CLI flag and set that in the test, or clean up the filesystem state (it seems we're already cleaning up the K8s state). cc @3u13r 

Long-term, I would still argue we should have a test framework for simulating IPCache delays in an artificial manner, I think that would also be useful for other kinds of tests as well.
